src/main/java/io/vertx/core/http/impl/Http1xServerRequest.java

```diff
@@ -123,18 +123,18 @@ void setRequest(HttpRequest request) {
     }
   }
 
-  private InboundBuffer<Object> pendingQueue() {
-    if (pending == null) {
-      pending = new InboundBuffer<>(context, 8);
-      pending.drainHandler(v -> conn.doResume());
-      pending.handler(buffer -> {
-        if (buffer == InboundBuffer.END_SENTINEL) {
-          onEnd();
-        } else {
-          onData((Buffer) buffer);
-        }
-      });
-    }
+  private InboundBuffer createAndSetInboundBuffer() {
+    assert pending == null;
```

Comment:
that change does not seem necessary for this PR

Reply:
what is the benefit of changing this ? am I missing something ?

Reply:
Nope it is related, but probably worth more information to be understood..
In the rest of the method you can find

```java
  private InboundBuffer createAndSetInboundBuffer() {
    assert pending == null;
    InboundBuffer<Object> pending = new InboundBuffer<>(context, 8);
    this.pending = pending;
    pending.drainHandler(v -> conn.doResume());
    pending.handler(buffer -> {
      if (buffer == InboundBuffer.END_SENTINEL) {
        onEnd();
      } else {
        onData((Buffer) buffer);
      }
    });
    return pending;
  }
  ```
as you can see `pending` is using a local just created `InboundBuffer`, calling 2 synchronized methods:
- `pending.drainHandler`
- `pending.handler`

I've measured that these 2 synchronized methods would "merge" into a single acquire/release of monitor instead of 2, thanks to using a local variable (which is supposed to be stable, compared to an access field), see https://github.com/franz1981/java-puzzles/blob/8851ce838603a425f38400c2a0dee739f3ba6d58/src/main/java/red/hat/puzzles/concurrent/LockCoarsening.java#L194 which shows that it perform at twice the speed if compared to the existing one.